src/clients/mocks/MockSpokePoolClient.ts

```diff
@@ -33,8 +34,14 @@ export class MockSpokePoolClient extends SpokePoolClient {
   public numberOfDeposits = 0;
   public blocks: Record<number, Block> = {};
 
-  constructor(logger: winston.Logger, spokePool: Contract, chainId: number, deploymentBlock: number) {
-    super(logger, spokePool, null, chainId, deploymentBlock);
+  constructor(
+    logger: winston.Logger,
+    spokePool: Contract,
+    chainId: number,
+    deploymentBlock: number,
+    opts: { hubPoolClient: HubPoolClient | null } = { hubPoolClient: null }
```

Comment:
Thoughts on making the HubPoolClient optional? It should have the same intended effect unless you explicitly need `null` instead of `undefined`.
```suggestion
    opts: { hubPoolClient?: HubPoolClient } = {}
```

Reply:
It specifically needs to be null per SpokePoolClient definition: https://github.com/across-protocol/sdk-v2/blob/b26b1feacf0c00e7ed13ed4303faf618790207ec/src/clients/SpokePoolClient.ts#L91.

We could change that if we want...

Reply:
Will merge this as-is and follow up out of band.